lib/matplotlib/inset.py

```diff
@@ -0,0 +1,258 @@
+        facecolor for the rectangle.
+
+        Parameters
+        ----------
+        c : :mpltype:`color`
+        """
+        self._shared_setter('edgecolor', c)
+        self._shared_setter('facecolor', c)
+
+    def set_linewidth(self, w):
+        """
+        Set the linewidth in points of the rectangle and the connectors.
+
+        Parameters
+        ----------
+        w : float or None
+        """
+        self._shared_setter('linewidth', w)
+
+    def set_linestyle(self, ls):
+        """
+        Set the linestyle of the rectangle and the connectors.
+
+        ==========================================  =================
+        linestyle                                   description
+        ==========================================  =================
+        ``'-'`` or ``'solid'``                      solid line
+        ``'--'`` or  ``'dashed'``                   dashed line
+        ``'-.'`` or  ``'dashdot'``                  dash-dotted line
+        ``':'`` or ``'dotted'``                     dotted line
+        ``'none'``, ``'None'``, ``' '``, or ``''``  draw nothing
+        ==========================================  =================
+
+        Alternatively a dash tuple of the following form can be provided::
+
+            (offset, onoffseq)
+
+        where ``onoffseq`` is an even length tuple of on and off ink in points.
+
+        Parameters
+        ----------
+        ls : {'-', '--', '-.', ':', '', (offset, on-off-seq), ...}
+            The line style.
+        """
+        self._shared_setter('linestyle', ls)
+
+    def _bounds_from_inset_ax(self):
+        xlim = self._inset_ax.get_xlim()
+        ylim = self._inset_ax.get_ylim()
+        return (xlim[0], ylim[0], xlim[1] - xlim[0], ylim[1] - ylim[0])
+
+    def _update_connectors(self):
+        (x, y) = self._rectangle.get_xy()
+        width = self._rectangle.get_width()
+        height = self._rectangle.get_height()
+
+        existing_connectors = self._connectors or [None] * 4
+
+        # connect the inset_axes to the rectangle
+        for xy_inset_ax, existing in zip([(0, 0), (0, 1), (1, 0), (1, 1)],
+                                         existing_connectors):
+            # inset_ax positions are in axes coordinates
+            # The 0, 1 values define the four edges if the inset_ax
+            # lower_left, upper_left, lower_right upper_right.
+            ex, ey = xy_inset_ax
+            if self.axes.xaxis.get_inverted():
+                ex = 1 - ex
+            if self.axes.yaxis.get_inverted():
+                ey = 1 - ey
+            xy_data = x + ex * width, y + ey * height
+            if existing is None:
+                # Create new connection patch with styles inherited from the
+                # parent artist.
+                p = ConnectionPatch(
+                    xyA=xy_inset_ax, coordsA=self._inset_ax.transAxes,
+                    xyB=xy_data, coordsB=self.axes.transData,
+                    arrowstyle="-",
```

Comment:
`zorder` disappeared from the original?

Reply:
Yes I originally had _zorder_ here but then got it in my head that only the zorder of the main artist would matter.  Basically I have no clue how/where _zorder_ ultimately gets applied.

Reply:
No, you are right, actually; `zorder` is sorted by the parent (`Axes` or `(Sub)Figure`), but these sub-artists aren't added to the `Axes`. Theoretically, the `InsetIndicator` could sort them by `zorder` internally, but they'd never go anywhere outside of its depth.